lib/commands/exam.js

```diff
@@ -124,35 +135,89 @@ module.exports = TestCommand.extend({
    * @override
    */
   _generateCustomConfigs: function(commandOptions) {
-    var config = this._super._generateCustomConfigs.apply(this, arguments);
-
-    if (this.validator.shouldParallelize) {
-      // Get the testPage from the generated config or the Testem config and
-      // use it as the baseUrl for the parallelized test pages
-      var baseUrl = config.testPage || this._getTestPage(commandOptions.configFile);
-      var count = commandOptions.split;
-      var partitions = commandOptions.partition;
-
-      if (!partitions) {
-        partitions = [];
-        for (var i = 0; i < commandOptions.split; i++) {
-          partitions.push(i + 1);
-        }
-      }
+    const config = this._super._generateCustomConfigs.apply(this, arguments);
+    const customBaseUrl = this._getCustomBaseUrl(config, commandOptions);
+    const shouldParallelize = this.validator.shouldParallelize;
+    const shouldLoadBalance = this.validator.shouldLoadBalance;
 
-      if (Array.isArray(baseUrl)) {
-        var command = this;
-        config.testPage = baseUrl.reduce(function(testPages, baseUrl) {
-          return testPages.concat(command._generateTestPages(baseUrl, partitions, count));
-        }, []);
-      } else {
-        config.testPage = this._generateTestPages(baseUrl, partitions, count);
+    if (!shouldParallelize && !shouldLoadBalance) {
+      return config;
+    }
+
+    let ids = commandOptions.partition;
+    let appendingParam = '_partition';
+
+    if (shouldParallelize && !ids) {
+      ids = [];
+      for (var i = 0; i < commandOptions.split; i++) {
+        ids.push(i + 1);
       }
+    } else if (shouldLoadBalance) {
+      config.additionalBrowserSocketEvents = this._addBrowserSocketEventsToConfig(config);
+      const browserCount = commandOptions.loadBalance;
+      appendingParam = '_browser';
+
+      ids = Array.from({length: browserCount}, (v,k) => k+1); // if browser=3 then id = [1,2, 3]
```

Comment:
I get what you're trying to say with this comment, but perhaps you can make it clearer? If you provide a value for browser, id becomes a range from 1..<browser>.